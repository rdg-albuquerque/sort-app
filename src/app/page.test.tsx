import "@testing-library/jest-dom"
import { render } from "@testing-library/react"
import Home from "@/app/page"
import { RecoilRoot } from "recoil"

const pushMock = jest.fn()
jest.mock("next/navigation", () => {
  return {
    useRouter: () => ({
      push: pushMock,
    }),
  }
})

describe("Homepage", () => {
  test("renders homepage correctly", () => {
    const { container } = render(
      <RecoilRoot>
        <Home />
      </RecoilRoot>
    )

    expect(container).toMatchSnapshot()
  })
})
