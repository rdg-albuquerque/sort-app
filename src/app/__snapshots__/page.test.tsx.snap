// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders homepage correctly 1`] = `
<div>
  <div
    class="flex min-h-screen flex-col items-center gap-24 p-24"
  >
    <form
      class="flex flex-col gap-2"
    >
      <input
        name="participant"
        placeholder="Insert the participants names"
        type="text"
        value=""
      />
      <button
        class="bg-gray-500 text-white px-[40px] py-[18px] rounded-[38px]"
        disabled=""
        type="submit"
      >
        Submit
      </button>
    </form>
    <ul>
      <li
        class="text-white"
      >
        Rodrigo
      </li>
      <li
        class="text-white"
      >
        Jo√£o
      </li>
      <li
        class="text-white"
      >
        Maria
      </li>
      <li
        class="text-white"
      >
        Ana
      </li>
    </ul>
    <div>
      <button
        class="bg-[#4646bb] text-white px-[40px] py-[18px] rounded-[38px]"
      >
        Iniciar brincadeira
      </button>
    </div>
  </div>
</div>
`;
